public/app/features/dashboard/timeSrv.js

```diff
@@ -75,7 +75,12 @@ define([
     };
 
     this.setAutoRefresh = function (interval) {
-      this.dashboard.refresh = interval;
+      if ($routeParams.refresh) {
```

Comment:
this does not look right as this method is called every time when refresh option is changed. The routeParams property should only be checked once on dashboard init, https://github.com/xginn8/grafana/blob/870c5f8cef172a9c5a8ea3493631d2bf47fdade7/public/app/features/dashboard/timeSrv.js#L65